AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs

```diff
@@ -79,13 +79,23 @@ private async Task DoSendMqttMessage(string topic, byte[] payload, int qos)
                                 .WithRetainFlag(false)
                                 .Build();
 
-            MqttClientPublishResult result = await _client.PublishAsync(applicationMessage);
+            try
+            {
+
+                MqttClientPublishResult result = await _client.PublishAsync(applicationMessage);
+
+                if (result.ReasonCode != MqttClientPublishReasonCode.Success)
+                {
+                    throw new AstarteTransportException
+                    ($"Error publishing on MQTT. Code: {result.ReasonCode}");
+                }
 
-            if (result.ReasonCode != MqttClientPublishReasonCode.Success)
+            }
+            catch (MqttCommunicationException)
             {
-                throw new AstarteTransportException
-                ($"Error publishing on MQTT. Code: {result.ReasonCode}");
+                _astarteTransportEventListener?.OnTransportDisconnected();
```

Comment:
I'm not sure that if `MqttCommunicationException` raise up, disconnecting the transport is a right behavior, maybe is better to propagate the exception.

Reply:
I put this line before init because I need to check if device.key is expired first. Init function read device.key from file if device.key PublishAsync method throw exception and agine call Connect method. Based on my test, this is a better way to regenerate a device.key.